engine/src/main/java/io/zeebe/engine/processing/bpmn/container/ProcessProcessor.java

```diff
@@ -70,67 +70,62 @@ public void onActivate(
   }
 
   @Override
-  @Override
-  public void onCompleting(
+  public void onComplete(
       final ExecutableFlowElementContainer element, final BpmnElementContext context) {
 
     eventSubscriptionBehavior.unsubscribeFromEvents(context);
-    stateTransitionBehavior.transitionToCompleted(context);
-  }
-
-  @Override
-  public void onCompleted(
-      final ExecutableFlowElementContainer element, final BpmnElementContext context) {
 
     final var parentProcessInstanceKey = context.getParentProcessInstanceKey();
+    BpmnElementContext parentElementInstanceContext = null;
     if (parentProcessInstanceKey > 0) {
-      // process instance is created by a call activity
-      stateTransitionBehavior.beforeExecutionPathCompleted(element, context);
-      stateTransitionBehavior.afterExecutionPathCompleted(element, context);
+      // needs to be done before we delete the Process element instance
+      parentElementInstanceContext = stateBehavior.getParentElementInstanceContext(context);
     }
 
-    if (element.hasNoneStartEvent()) {
```

Comment:
Is this check not necessary anymore?

Reply:
it was also before not really necessary, we will not have any metadata for it to send the result